### PR TITLE
lint: Remove special-case for `react-hooks/exhaustive-deps` in CI rules

### PR DESCRIPTION
**Pull request by ryan953**
This popped up as an issue in CI. I'm ignoring the warning because the team selector is a bit complex, and can have long lists. I'd rather not break it by adding dependencies and causing additional unexpected re-renders.

So just silencing the warning for now to make CI happy.

### CODE REVIEW

File: static/app/components/teamSelector.tsx [hunk 1]
```diff
@@ -377,7 +377,7 @@ function TeamSelector(props: Props) {
     }
     // We only want to do this once when the component is finished loading for teams and mounted.
     // If the user decides they do not want the default, we should not add the default value back.
-  }, [fetching, useTeamDefaultIfOnlyOne]);
+  }, [fetching, useTeamDefaultIfOnlyOne]); // eslint-disable-line react-hooks/exhaustive-deps
```
Comment (evanpurkhiser) [inline static/app/components/teamSelector.tsx:380]: I don't get it, why is this an error in CI all of a sudden?

Comment (ryan953) [reply static/app/components/teamSelector.tsx:380]: Not sure, CI was grumpy over here: https://sentry.slack.com/archives/C8V02RHC7/p1710193569187799

Comment (ryan953) [reply static/app/components/teamSelector.tsx:380]: This statement is missing dependencies so it should complain

Comment (evanpurkhiser) [reply static/app/components/teamSelector.tsx:380]: Yes but we have the warning silenced in CI do we not?

**Review by gggritso — APPROVED** [review]
Spooky CI errors

File: static/app/components/teamSelector.tsx (hunk 1 above)
Comment (evanpurkhiser) [reply static/app/components/teamSelector.tsx:380]: https://github.com/getsentry/sentry/blob/e7683c892593b163eb35c81c6f073598dc0dfebd/.eslintrc.js#L10-L15

Comment (ryan953) [reply static/app/components/teamSelector.tsx:380]: oh thats true... but this is the last one that popped up when i was manually trying to findout what happened in that chat thread... so maybe we should enable the rule on CI so we dont' add more exhaustive-deps problems going forward! i didn't think of that before, but it'll be good

Comment (evanpurkhiser) [reply static/app/components/teamSelector.tsx:380]: Oh nice, seems like we fixed all of them already.

I wonder if that's true in getsentry as well.

Let's remove this from the eslint config and move it into our config repo?

**Review by evanpurkhiser — APPROVED** [review]

**Pull request merged**